### PR TITLE
Merge series/1.x to series/2.x

### PR DESCRIPTION
**Pull request by sideeffffect**
Please merge this as a _Merge_ merge (not as Squash merge).

### CODE REVIEW

File: core/shared/src/main/scala/zio/Chunk.scala [hunk 1]
```diff
@@ -46,47 +46,53 @@ sealed abstract class Chunk[+A] extends ChunkLike[A] with Serializable { self =>
...
-        if (math.abs(diff) <= 1) Chunk.Concat(self, that)
-        else if (diff < -1) {
-          if (self.left.depth >= self.right.depth) {
-            val nr = self.right ++ that
-            Chunk.Concat(self.left, nr)
-          } else {
-            val nrr = self.right.right ++ that
-            if (nrr.depth == self.depth - 3) {
-              val nr = Chunk.Concat(self.right.left, nrr)
+    if (isEmpty) {
```
Comment (sideeffffect) [inline core/shared/src/main/scala/zio/Chunk.scala]: @shawjef3 I see suspicious changes in `Chunk`.
Is this the intended content of `Chunk` for ZIO 2?
Wasn't this already merged in https://github.com/zio/zio/pull/7400 ?

Comment (adamgfraser) [reply core/shared/src/main/scala/zio/Chunk.scala]: @sideeffffect Please don't merge these changes into `series/2.x`.

File: core/shared/src/main/scala/zio/Chunk.scala [hunk 2]
```diff
@@ -1375,7 +1381,7 @@ object Chunk extends ChunkFactory with ChunkPlatformSpecific {
       case x: AppendN[_]            => x.classTag.asInstanceOf[ClassTag[A]]
       case x: Arr[_]                => x.classTag.asInstanceOf[ClassTag[A]]
       case x: Concat[_]             => x.classTag.asInstanceOf[ClassTag[A]]
-      case Empty                    => classTag[java.lang.Object].asInstanceOf[ClassTag[A]]
+      case _: Empty.type            => classTag[java.lang.Object].asInstanceOf[ClassTag[A]]
```
Comment (adamgfraser) [inline core/shared/src/main/scala/zio/Chunk.scala]: These changes are also not necessary.

File: core/shared/src/main/scala/zio/Chunk.scala (hunk 1 above)
Comment (sideeffffect) [reply core/shared/src/main/scala/zio/Chunk.scala]: Alright, no problem.

**Review by adamgfraser — APPROVED** [review]

**Comment by sideeffffect** [comment]
Can we restart just that one failing test?

**Comment by adamgfraser** [comment]
Yes will do when remaining tests have completed.

**Comment by sideeffffect** [comment]
Completed now :+1:

**Comment by sideeffffect** [comment]
Mergable :tada:

**Comment by adamgfraser** [comment]
Thank you! 🙏

**Pull request merged**